### DISCUSSION TITLE
Typescript calls between getters

### DISCUSSION THREAD

**Discussion by alex129** [vuejs/pinia#2149] [Help and Questions] [answered]
Hi, I have a store and I want to call getter inside an other getter, like that:

![image](https://user-images.githubusercontent.com/10452211/233006722-82083199-bbeb-4207-9cb4-3630ea8c48ee.png)

The problem is that if I use `this` context, typescript detect is a function so when I try to acces the properties or the returned object I can't cause it's a function, I could set the types like as Type but typescript forces to me to use as unknown as Type or something like that

Other example

![image](https://user-images.githubusercontent.com/10452211/233007343-76d1af3b-f3ff-4609-9e28-04ffc1a94f9e.png)

Any idea?

**Comment by posva** [comment, marked as answer]
Add the return type: https://pinia.vuejs.org/core-concepts/getters.html#getters

**Reply by alex129** [reply]
Sigue pasando:

![image](https://user-images.githubusercontent.com/10452211/233055151-4249e8b5-b907-4a2f-a888-46b9af3175ca.png)

**Reply by posva** [reply]
Es el otro que tienes que tipar.

**Reply by alex129** [reply]
Si, lo he probado incluso el mismo ejemplo que en la docu. El problema es que tengo los getters fuera del define. He puesto este ejemplo pero realmente tengo varias stores con los getters separados. Actualmente hemos resuelto el inconveniente de llamar entre getters con esto:

  `      const store = this as unknown as ReturnType<typeof useCoreStore>;`

De esta manera puedo tener algo como esto:

![image](https://user-images.githubusercontent.com/10452211/233066359-1b915ae7-e1a5-4899-960c-db7a953a1bd2.png)

Pero es algo que no me gusta hacer, me gustaria encontrar una manera de que typescript interpretase esos getters fuera del define y poder tenerlo en archivos separados. He pensado en hacerlo con composition pero me gusta el Options API para las stores
